File: src/main/java/me/leodev/trollingfreedomreborn/other/EventListener.java
package me.leodev.trollingfreedomreborn.other;

import me.leodev.trollingfreedomreborn.commands.UnTroll;
import me.leodev.trollingfreedomreborn.main.Core;
import com.cryptomorin.xseries.XMaterial;
import me.leodev.trollingfreedomreborn.ui.PlayerSelectorInventory;
import me.leodev.trollingfreedomreborn.ui.TrollInventory;
import org.bukkit.entity.Item;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.block.Action;
import org.bukkit.event.entity.EntityDamageEvent;
import org.bukkit.event.player.PlayerDropItemEvent;
import org.bukkit.event.player.PlayerInteractEvent;
import org.bukkit.event.player.PlayerMoveEvent;
import org.bukkit.event.player.PlayerQuitEvent;

import java.io.IOException;


public class EventListener implements Listener {


    public String getNOU(Player p) {
        return Core.uid() ? p.getName() : p.getUniqueId().toString();
    }

    @org.bukkit.event.EventHandler
    public void onPlayerFall(EntityDamageEvent e) {
        if (e.getEntity() instanceof Player) {
            Player player = (Player) e.getEntity();
            try {
                if (e.getCause() == EntityDamageEvent.DamageCause.FALL) {
                    if (TrollInventory.getMaps("LP").containsKey(Core.instance.getConfig().getBoolean("using-uuid") ? player.getName() : player.getUniqueId().toString())) {
                        e.setCancelled(true);
                        TrollInventory.getMaps("LP").remove(Core.instance.getConfig().getBoolean("using-uuid") ? player.getName() : player.getUniqueId().toString());
                    }
                }
            } catch (Exception ignored) {
            }
        }
    }

    @EventHandler
    public void onPlayerInteract(PlayerInteractEvent e) {
        Player p = e.getPlayer();
        if (Core.advCheck("ms3.use", p)) {
            if (e.getAction() == Action.RIGHT_CLICK_BLOCK || e.getAction() == Action.RIGHT_CLICK_AIR) {
                if (p.getInventory().getItemInMainHand() != null && p.getInventory().getItemInMainHand().getType() != XMaterial.AIR.parseMaterial()) {
                    if (p.getInventory().getItemInMainHand().isSimilar(Core.instance.getSkull())) {
                        PlayerSelectorInventory ps = new PlayerSelectorInventory();
                        ps.openSel(p);
                        e.setCancelled(true);

                    }
                }
            }
        }
    }

    @EventHandler
    public void onPlayerDisconnect(PlayerQuitEvent e) throws IOException {
        Player p = e.getPlayer();
        UnTroll stoptroll = new UnTroll();
        stoptroll.StopTrolls(p, p);
    }

    @EventHandler
    public void onMove(PlayerMoveEvent e) {
        if (e.getPlayer() instanceof Player) {
            Player player = e.getPlayer();
            try {
                if (TrollInventory.getMaps("FR").containsKey(Core.uid() ? player.getName() : player.getUniqueId().toString())) {
                    e.setCancelled(true);
                }
            } catch (Exception ignored) {
            }
        }
    }

    @EventHandler
    public void onPlayerDrop(PlayerDropItemEvent e) {
        Player p = e.getPlayer();
        if (Core.advCheck("trollingfreedom.troll", p)) {
            Item droppedi = e.getItemDrop();
            if (droppedi.getItemStack().isSimilar(Core.instance.getSkull())) {
                p.sendMessage(Core.instance.getP() + "This item is undroppable.");
                e.setCancelled(true);
            }
        }
    }

}

File: src/main/java/com/leomadrassi/trollingfreedomreborn/other/EventListener.kt
package com.leomadrassi.trollingfreedomreborn.other

import com.leomadrassi.trollingfreedomreborn.commands.UnTroll
import com.leomadrassi.trollingfreedomreborn.main.Core
import com.cryptomorin.xseries.XMaterial
import com.leomadrassi.trollingfreedomreborn.ui.PlayerSelectorInventory
import com.leomadrassi.trollingfreedomreborn.ui.TrollInventory
import org.bukkit.entity.Item
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.block.Action
import org.bukkit.event.entity.EntityDamageEvent
import org.bukkit.event.player.PlayerDropItemEvent
import org.bukkit.event.player.PlayerInteractEvent
import org.bukkit.event.player.PlayerMoveEvent
import org.bukkit.event.player.PlayerQuitEvent
import java.io.IOException

class EventListener : Listener {

    fun getNOU(p: Player): String {
        return if (Core.uid()) p.name else p.uniqueId.toString()
    }

    @EventHandler
    fun onPlayerFall(e: EntityDamageEvent) {
        if (e.entity is Player) {
            val player = e.entity as Player
            try {
                if (e.cause == EntityDamageEvent.DamageCause.FALL) {
                    val key = if (Core.instance.config.getBoolean("using-uuid")) player.name else player.uniqueId.toString()
                    if (TrollInventory.getMaps("LP")?.containsKey(key) == true) {
                        e.isCancelled = true
                        TrollInventory.getMaps("LP")?.remove(key)
                    }
                }
            } catch (_: Exception) {
            }
        }
    }

    @EventHandler
    fun onPlayerInteract(e: PlayerInteractEvent) {
        val p = e.player
        if (Core.advCheck("ms3.use", p)) {
            if (e.action == Action.RIGHT_CLICK_BLOCK || e.action == Action.RIGHT_CLICK_AIR) {
                if (p.inventory.itemInMainHand != null && p.inventory.itemInMainHand.type != XMaterial.AIR.parseMaterial()) {
                    if (p.inventory.itemInMainHand.isSimilar(Core.instance.getSkull())) {
                        PlayerSelectorInventory().openSel(p)
                        e.isCancelled = true
                    }
                }
            }
        }
    }

    @Throws(IOException::class)
    @EventHandler
    fun onPlayerDisconnect(e: PlayerQuitEvent) {
        val p = e.player
        UnTroll().StopTrolls(p, p)
    }

    @EventHandler
    fun onMove(e: PlayerMoveEvent) {
        if (e.player is Player) {
            val player = e.player
            try {
                val key = if (Core.uid()) player.name else player.uniqueId.toString()
                if (TrollInventory.getMaps("FR")?.containsKey(key) == true) {
                    e.isCancelled = true
                }
            } catch (_: Exception) {
            }
        }
    }

    @EventHandler
    fun onPlayerDrop(e: PlayerDropItemEvent) {
        val p = e.player
        if (Core.advCheck("trollingfreedom.troll", p)) {
            val droppedi = e.itemDrop
            if (droppedi.itemStack.isSimilar(Core.instance.getSkull())) {
                p.sendMessage(Core.instance.getP() + "You cant drop this item.")
                e.isCancelled = true
            }
        }
    }
}
